### ISSUE TITLE
Main window & Miniplayer restores offscreen when using multiple monitors / Disconnecting

### ISSUE THREAD

**Issue by Smity** [ytmdesktop/ytmdesktop#918]
### Consent

- [X] I verified that there is no open issue for the same subject.
- [X] I understand that **YTMDesktop have NO affiliation with Google or YouTube**

### Current Behavior

Using the miniplayer on one screen works fine. I can switch between mini and full modes, and both modes remember their respective window size settings. (If the full mode was taking up most of the left side of my screen before switching to the miniplayer, it returns to the left side when I switch modes; likewise, if the miniplayer was in the lower-right corner of the screen when I switched to full mode, it returns to the lower-right when switching back.) However, this only works on the MAIN monitor (whichever is set to main in Windows's display settings).

If I'm using the miniplayer on my secondary monitor and I expand to the full window, then return to miniplayer-mode, the miniplayer window moves offscreen. I have to use ALT+Space to maximize the window, which moves it to my main monitor; then I can drag it to my secondary monitor. This seems to only affect the miniplayer. The full window always returns to the correct position and size regardless of which monitor it was on.

### Expected Behavior

The miniplayer should return to its last position and size every time it is activated, or a similar position on the main monitor if the monitor it was originally placed on has been disconnected.

### Steps To Reproduce

1) Open the miniplayer.
2) Move the miniplayer the the secondary monitor.
3) Toggle to full-mode, then back to miniplayer-mode.
The miniplayer will not be visible on either screen.

4) Press ALT-Space to open a context menu.
5) Click "Maximize", or press x.
The miniplayer will be restored to full screen on the main monitor.

### YTMDesktop

1.13.0

### OS

Windows

### OS Version

10

### Arch

x64

### Installation way

.exe

### Anything else?

My secondary monitor is placed above my main monitor; not beside it, which is the typical arraignment.

**Comment by Alipoodle** [comment]
1.14 has some changes which improve upon this, but it's not 100% perfect.
If you have entirely lost them go to the `%appdata%/youtube-music-desktop-app` and open the `config.json` and edit the X/Y position values for Miniplayer to be 100.

**Comment by PATAPOsha** [comment]
> 

thx
I can confirm same issue on Windows. Except I have only one monitor, 34'' 3440x1440. Sometimes miniplayer goes offscreen. Changes in `AppData\Roaming\youtube-music-desktop-app\config.json` helped.
```
"miniplayer-position": {
		"x": 100,
		"y": 100
	}
```

**Comment by PATAPOsha** [comment]
@Alipoodle heppens every time after restart for me. Maybe set  X/Y=100 as default? Without saving previous position? Or at least with ability to disbable it.

**Comment by rdennis** [comment]
I've had this issue (all YTMD windows opened off screen) and I wasn't able to move them in any way. What did work was using `Win + D` twice to show/hide the desktop and suddenly my YTMD windows were positioned on screen 🤷

**Comment by Alipoodle** [comment]
that's a good thing to know.
I'll test that if I get that again, but yea, a full on fix is needed to be implemented 😅

**Comment by milesbuckton** [comment]
I had the same problem with versions `1.13.0` - `1.14.2`. My issue was working with dual monitors, and the window would go off of the screen for some strange reason. I resolved it by switching to either one monitor, or changing to a higher resolution than what it was set to originally.

I reset the `window-position`, and `miniplayer-position` coordinates to `0` in the `%appdata%/youtube-music-desktop-app\config.json` file. This resolved my reoccuring issue! 👏🏻 

```JSON
	"window-position": {
		"x": 0,
		"y": 0
	},
		"miniplayer-position": {
		"x": 0,
		"y": 0
	},
```

**Comment by guywmustang** [comment]
https://github.com/ytmdesktop/ytmdesktop/pull/1076 - My PR will fix this for the miniplayer.

**Comment by amenoko** [comment]
So when I edit config.json and set xy to 100, 100 the window spawns on my primary monitor just fine. If i don't move it, close it, reopen, it'll spawn in the same place. However, if I move it to my secondary monitor, close, reopen, suddenly it'll insist on spawning in really strange places. If I tile it to the left half of the monitor, it'll save x -1920 y 72. If I tile it to the right half, it'll save as -960, 72. I can move it to the main monitor (Alt+Space, Win+Arrows etc), close, and it'll reopen in the same place/size as when I closed it. This is on a fresh git clone, although the issue is identical on the recent nightly build too.

**Comment by makisukurisu** [comment]
Sadly, this is still the case with 2.0 RC5 version, and it affects main window, with no other option but to change the state in config file. (Or connect the second monitor)
![image](https://github.com/ytmdesktop/ytmdesktop/assets/49622129/1b339ab0-b75d-4c43-82be-0b3be9c53772)

Should this be reported separately, or can you add a label for v2?

**Comment by Alipoodle** [comment]
> Sadly, this is still the case with 2.0 RC5 version, and it affects main window, with no other option but to change the state in config file. (Or connect the second monitor)
> 
> Should this be reported separately, or can you add a label for v2?

I've been rolling with V2 for over 5 months now and I swap between multiple monitors setups and have yet to counter this bug 😅
It's indeed worth having a new issue for this, but we won't handle windows in the same way anymore, so I'm curious to know what has happened

**Comment by makisukurisu** [comment]
![image](https://github.com/ytmdesktop/ytmdesktop/assets/49622129/b8748e4a-8f5e-43c6-9e52-f1c60cda9ccd)
![image](https://github.com/ytmdesktop/ytmdesktop/assets/49622129/0c8310d4-24a1-41c7-8eda-530836dcc91a)

I suppose it may be related to how Windows tells you the windows bounds location.

I have a two monitor setup, the external monitor is number 2, and I set it as my Main monitor (So, I assume it has the 0, 0 point in the top left corner).
YouTube Music app is located in the left corner of my first display (which is my laptops screen).
So, then I had to disconnect the external display, and I was left with only the first monitor.
And since it had the negative X, I assume that the app was sent all the way to the left, outside my screen's bounds.

**Comment by Alipoodle** [comment]
Closing issue due to no longer supporting YouTube Music Desktop v1 + Release of YouTube Music v2.
Please re-open this issue if this is still occurring in the latest build of the application, however as a re-write has been done on the entire project most issues will be fixed, no longer a feature or re-worked.

**Comment by makisukurisu** [comment]
This is still happening on my machine, with the newly released V2 (commit 6d0abf5)

The problem will not happen if you use the app in full-screen. I can try to record the issue, perhaps it'll make it clear how to reproduce it, and, in turn — fix the bug.
I've uploaded the recordings to [youtube (in processing at the moment of sending this comment)](https://www.youtube.com/watch?v=llA-qHwC2U4) and to [OneDrive (so you can download it)](https://1drv.ms/v/s!AqrlfR_sZDUhtFw1DFfqpDpZyeEx?e=0gRwps).

If you need more explanations from my side, I'll gladly help.

Note — I've used PowerToys Fancy Zones to change the size of the app, but it doesn't matter how exactly you change the size, so long as it's not in full screen.

**Comment by Alipoodle** [comment]
> This is still happening on my machine, with the newly released V2 (commit [6d0abf5](https://github.com/ytmdesktop/ytmdesktop/commit/6d0abf5c658b4e8e91c256bf5db9d041cf203255))
> 
> The problem will not happen if you use the app in full-screen. I can try to record the issue, perhaps it'll make it clear how to reproduce it, and, in turn — fix the bug. I've uploaded the recordings to [youtube (in processing at the moment of sending this comment)](https://www.youtube.com/watch?v=llA-qHwC2U4) and to [OneDrive (so you can download it)](https://1drv.ms/v/s!AqrlfR_sZDUhtFw1DFfqpDpZyeEx?e=0gRwps).
> 
> If you need more explanations from my side, I'll gladly help.
> 
> Note — I've used PowerToys Fancy Zones to change the size of the app, but it doesn't matter how exactly you change the size, so long as it's not in full screen.

Could you create a new Issue with this information.
considering it's to do with NOT being in full screen it's a bit easier to indicate where the issue is coming from~ however makes it a bit lower of a priority of a bug.

**Comment by Rcmanaka** [comment]
Sadly this is happenning in YTMD v 1.14.2 running on Windows 11 pro 23H2 build 22631.4112 and the Windows Feature Experience Pack 1000.22700.1034.0.  My left monitor is 2560 x 14440 (used as main display) and my right monitore is 3840 x 2160.  I also am using Display Fusion and my display connection is through an NVIDIA GeForce RTX 2060 Super.

YTMD runs fine on my left monitor (although the mini player icon disappeared from the bottom control bar) but when I switch on the mini-player using the shortcut keys and switch the player to my right monitor the mini-player "hunts" to rescale the player and choses a random size and location.  If I switch the mini-player back to the second monitor it works correctly!

If I move the full desktop player to the right monitor and THEN switch to the mini-player then the mini-player seems to size, locate and work correctly.

HTH!
